docs/index.rst

```diff
@@ -1,14 +1,207 @@
-* Automatic creation and update of Requirements and Test Cases from a Python
+.. _what_is_betelgeuse:
+
+What is Betelgeuse?
+```````````````````
+
+Betelgeuse is a python program that reads standard Python test cases and offers
+tools to interact with Polarion. Possible interactions are:
+
+* Automatic creation/updation of Requirements and Test Cases from a Python
   project code base.
-* Creation of Test Runs based on a jUnit XML file.
+* Automatic Creation/updation of Test Runs based on a jUnit XML file.
+
+Betelgeuse uses `Testimony <https://pypi.python.org/pypi/testimony>`_ and
+Pylarion projects to parse test cases and communicate with polarion
+respectively.
+
+.. _prerequisites:
+
+Prerequisites
+`````````````
+Login to polarion in a browser to check if the login user has permissions to
+create/update the following entities in Polarion:
+
+* requirement
+* test case
+* test run
+
+.. _quick_start:
+
+Quick Start
+```````````
+1. Betelgeuse uses Pylarion to interact with polarion. Install Pylarion from its
+   source.
+
+   .. note::
+
+     - It may be possible that the latest version of Betelgeuse may not work
+       correctly with some versions of Pylarion.  In this case, please use an
+       alternate working version of Pylarion.
+     - Read Pylarion documentation and set up ``.pylarion`` config file as
+       required.
+
+2. Install betelguese from pypi.
+
+   .. code-block:: bash
+
+       $ pip install betelgeuse
+
+3. Alternatively you can install from source:
+
+   .. code-block:: bash
+
+       $ git clone https://github.com/SatelliteQE/betelgeuse.git 
+       $ cd betelgeuse
+       $ pip install -e .
+
+   .. note:: It is always recommended to use python virtual environment
+
+.. _usage_examples:
+
+Usage Examples
+``````````````
+.. note::
+
+  1. For easy understanding of Betelgeuse, this repository is already included with
+  ``sample_project`` folder.  This folder contains sample tests and XML results which
+  will help in setting up and testing Betelgeuse for your project.  The sample
+  commands used below also use this data.
+
+  2. Always run the test runner and Betelgeuse on the same directory to make
+  sure that the test run ID mapping works fine.  Otherwise Betelgeuse may
+  report ID errors.  More info in test_run_command_ section
+
+.. _help_command:
+
+help command
+++++++++++++
+
+.. code-block:: bash
+
+    $ betelgeuse --help
+
+.. _test_case_command:
+
+test-case command
++++++++++++++++++
+Creates/Updates test cases in polarion. This command performs the following
+steps:
+
+- Testimony is called to parse the test cases.
+- For each parsed test case, the following actions are performed:
+
+    - If ``@requirement`` token is present in the test case, it will be used as the
+      requirement name.  Otherwise it is derived from the test module name.  For
+      example, if the test module name is ``test_login_example``, then the
+      requirement name is ``Login Example``.
+    - Requirement is created.
```

Comment:
First, we try to find the requirement by its name. If nothing is found, then we try to create one.